scanner/src/main/kotlin/Scanner.kt

```diff
@@ -79,6 +79,15 @@ class Scanner(
         require(scannerWrappers.isNotEmpty() && scannerWrappers.any { it.value.isNotEmpty() }) {
             "At least one ScannerWrapper must be provided."
         }
+
+        scannerWrappers.values.flatten().distinct().forEach { scannerWrapper ->
+            scannerWrapper.criteria?.let { criteria ->
+                require(criteria.matches(scannerWrapper.details)) {
+                    "The scanner details of scanner '${scannerWrapper.details.name}' must satisfy the configured " +
+                        "criteria for looking up scan storage entries."
```

Comment:
It would be good to also print the configured criteria, I would suggest to use this:
`... do not satisfy the configured criteria '${criteria}'.

Reply:
I deliberately didn't do this as it is easy to look it up by scanner name.

Reply:
It would be even easier if the criteria was in the log message and does not have to be looked up at all. After all, this is not an error that should occur often so I'm not sure why concise is important.
In general I prefer if error messages describe the error ("does not satisfy") instead of the requirement ("must satisfy"), but that's the nit-pick part.

Reply:
No, I will not block this because of a nit pick.